### ISSUE TITLE
add perplexity metric to Bert model

### ISSUE THREAD

**Issue by mapingshuo** [PaddlePaddle/PaddleFleetX#281]
```python
import numpy as np
import fleetx as X
import paddle
import paddle.fluid as fluid
import paddle.distributed.fleet as fleet
import paddle.distributed.fleet.base.role_maker as role_maker
import time

# FleetX help users to focus more on learning to train a large scale model
# if you want to learn how to write a model, fleetx is not for you
# focus more on engineering staff in fleet-x

configs = X.parse_train_configs()
batch_size=53
configs.lr = 1e-4

role = role_maker.PaddleCloudRoleMaker(is_collective=True)
fleet.init(role)
# load Bert_large / Bert_base model
model = X.applications.BertLarge()
data_loader = model.load_digital_dataset_from_file(
    data_dir='../../data/bert/train',
    vocab_path='uncased_L-24_H-1024_A-16/vocab.txt',
    max_seq_len=512,
    batch_size=batch_size,
    in_tokens=False,
)

place = fluid.CUDAPlace(int(os.environ.get('FLAGS_selected_gpus', 0)))
exec_strategy = fluid.ExecutionStrategy()
exec_strategy.num_threads = 2
exec_strategy.num_iteration_per_drop_scope = 1
dist_strategy = fleet.DistributedStrategy()
dist_strategy.execution_strategy = exec_strategy
dist_strategy.nccl_comm_num = 3
dist_strategy.recompute_configs = {"checkpoints": model.checkpoints}
dist_strategy.recompute = True
print("base lr: ", configs.lr)
scheduled_lr = X.utils.linear_warmup_decay(configs.lr, warmup_steps=4000,
                                               num_train_steps=1000000)
optimizer = fluid.optimizer.Adam(learning_rate=scheduled_lr)
optimizer = fleet.distributed_optimizer(optimizer, dist_strategy)

clip_norm_thres = 1.0
fluid.clip.set_gradient_clip(
    clip=fluid.clip.GradientClipByGlobalNorm(clip_norm=clip_norm_thres))

optimizer.minimize(model.loss)


trainer = X.MultiGPUTrainer()
trainer.fit(model, data_loader, epoch=10)
```

result:

```shell
worker_index: 0, step11, train_loss: 10.793158, total time cost = 9.053438, step per second: 0.110455, speed: 0.110455
worker_index: 0, step12, train_loss: 10.778522, total time cost = 17.794933, step per second: 0.112392, speed: 0.114397
worker_index: 0, step13, train_loss: 10.719225, total time cost = 27.523812, step per second: 0.108997, speed: 0.102787
worker_index: 0, step14, train_loss: 10.759233, total time cost = 36.688284, step per second: 0.109027, speed: 0.109117
worker_index: 0, step15, train_loss: 10.764440, total time cost = 46.571331, step per second: 0.107362, speed: 0.101183
worker_index: 0, step16, train_loss: 10.703083, total time cost = 56.291579, step per second: 0.106588, speed: 0.102878
worker_index: 0, step17, train_loss: 10.717890, total time cost = 64.846372, step per second: 0.107947, speed: 0.116894
```

expected result:

add perplexity metric
